delira/data_loading/sampler/weighted.py

```diff
@@ -0,0 +1,95 @@
+from delira.data_loading.sampler.abstract import AbstractSampler
+from delira.data_loading.dataset import AbstractDataset
+import numpy as np
+
+
+class WeightedRandomSampler(AbstractSampler):
+    """
+    Class implementing Weighted Random Sampling
+    """
+
+    def __init__(self, weights, num_samples=None):
+        """
+
+        Parameters
+        ----------
+        weights : list
+            per-sample weights
+        num_samples : int
+            number of samples to provide. If not specified this defaults to
+            the amount of values given in :param:`num_samples´
+        """
+        if num_samples is None:
+            num_samples = len(weights)
+
+        self._num_samples = num_samples
+        super().__init__(np.arange(num_samples))
+        self._weights = weights
+
+    def __iter__(self):
+        """
+        Defines the actual weighted random sampling
+
+        Returns
+        -------
+        Iterator
+            iterator producing random samples
+        """
+        return iter(np.random.choice(self._indices, size=self._num_samples,
+                                     p=self._weights))
+
+    def __len__(self):
+        """
+        Defines the length of the sampler
+
+        Returns
+        -------
+        int
+            the number of samples
+        """
+        return self._num_samples
```

Comment:
There might be a test missing for the length

Reply:
I don't think such a test is necessary here. We only return a single fixed value here

Reply:
I think, our goal should be 100% coverage. This also includes testing these small things. Other opinions @haarburger @ORippler ?